ksa_compliance/ksa_compliance/doctype/zatca_business_settings/zatca_business_settings.py

```diff
@@ -263,6 +276,52 @@ def on_trash(self) -> None:
         frappe.throw(msg=_("You cannot Delete a configured ZATCA Business Settings"),
                      title=_("This Action Is Not Allowed"))
 
+    def create_tax_account(self) -> str:
+        account_doc = frappe.new_doc("Account")
+        parent_account = frappe.get_value("Account", {"company": self.company, "account_name": "Duties and Taxes"}
+                                          , "name")
+        account_doc.parent_account = parent_account
+        account_doc.company = self.company
+        account_doc.account_name = self.account_name
+        account_doc.account_number = self.account_number
+        account_doc.account_type = "Tax"
+        account_doc.tax_rate = self.tax_rate
+        account_doc.insert(ignore_permissions=True)
+        self.linked_tax_account = account_doc.name
+        self.save(ignore_permissions=True)
```

Comment:
We shouldn't be ignoring permissions on `ZATCA Business Settings` itself. As noted above, if `before_insert` works for us, we can do away with this.